### ISSUE TITLE
Tripal 2 upgrade process fails

### ISSUE THREAD

**Issue by almasaeed2010** [tripal/tripal#904]
### System information

* Tripal Version: 7.x-2.x
* Drupal Version: 7.6
* PostgreSQL Version: 9.6
* PHP Version: 7.3

### Issue description

Starting from scratch with a new drupal site, installing tripal 2, installing chado and enabling all sub modules such as tripal_feature and tripal_stock then disabling tripal_core causes an error.

#### Steps to reproduce

- Create a new Drupal site
- Install tripal 7.x-2.x (from github directly)
- Install chado using the GUI
- Enable all sub modules: `drush pm-enable tripal_core, tripal_views, tripal_db, tripal_cv, tripal_analysis, tripal_organism, tripal_feature, tripal_pub, tripal_stock`
- Disable tripal_core: `drush dis -y tripal_core`

#### Error messages and screenshots

```
 drush dis -y tripal_core
The following extensions will be disabled: tripal_core, tripal_views, tripal_db, tripal_cv, tripal_analysis, tripal_contact, tripal_organism, tripal_feature, tripal_pub, tripal_stock
Do you really want to continue? (y/n): y
tripal_analysis was disabled successfully.                                                                     [ok]
tripal_contact was disabled successfully.                                                                      [ok]
tripal_core was disabled successfully.                                                                         [ok]
tripal_cv was disabled successfully.                                                                           [ok]
tripal_db was disabled successfully.                                                                           [ok]
tripal_feature was disabled successfully.                                                                      [ok]
tripal_organism was disabled successfully.                                                                     [ok]
tripal_pub was disabled successfully.                                                                          [ok]
tripal_stock was disabled successfully.                                                                        [ok]
tripal_views was disabled successfully.                                                                        [ok]
Unable to find a view by the name of 'tripal_stock_user_stocks'. Unable to disable this view.                  [error]
Unable to find a view by the name of 'tripal_stock_admin_stocks'. Unable to disable this view.                 [error]
Unable to find a view by the name of 'tripal_pub_admin_publications'. Unable to disable this view.             [error]
Unable to find a view by the name of 'tripal_feature_user_feature'. Unable to disable this view.               [error]
Unable to find a view by the name of 'tripal_feature_admin_features'. Unable to disable this view.             [error]
Unable to find a view by the name of 'tripal_organism_user_organisms'. Unable to disable this view.            [error]
Unable to find a view by the name of 'tripal_organism_admin_organisms'. Unable to disable this view.           [error]
Unable to find a view by the name of 'tripal_contact_user_contacts'. Unable to disable this view.              [error]
Unable to find a view by the name of 'tripal_contact_admin_contacts'. Unable to disable this view.             [error]
Unable to find a view by the name of 'tripal_analysis_user_analyses'. Unable to disable this view.             [error]
Unable to find a view by the name of 'tripal_analysis_admin_analyses'. Unable to disable this view.            [error]
Unable to find a view by the name of 'tripal_cv_admin_cvs'. Unable to disable this view.                       [error]
Unable to find a view by the name of 'tripal_cv_admin_cvterms'. Unable to disable this view.                   [error]
Unable to find a view by the name of 'tripal_db_admin_dbs'. Unable to disable this view.                       [error]
Unable to find a view by the name of 'tripal_db_admin_dbxrefs'. Unable to disable this view.                   [error]

drush en -y tripal
The following extensions will be enabled: tripal
Do you really want to continue? (y/n): y
DatabaseSchemaObjectExistsException: Table tripal_custom_quota       [error]
already exists. in /var/www/html/includes/database/schema.inc:663
Stack trace:
#0 /var/www/html/includes/database/database.inc(2776):
DatabaseSchema->createTable('tripal_custom_q...', Array)
#1 /var/www/html/includes/common.inc(7132):
db_create_table('tripal_custom_q...', Array)
#2 /var/www/html/includes/module.inc(483):
drupal_install_schema('tripal')
#3
/root/.composer/vendor/drush/drush/commands/core/drupal/environment_7.inc(143):
module_enable(Array)
#4 /root/.composer/vendor/drush/drush/commands/pm/pm.drush.inc(1181):
drush_module_enable(Array)
#5 /root/.composer/vendor/drush/drush/includes/command.inc(422):
drush_pm_enable('tripal')
#6 /root/.composer/vendor/drush/drush/includes/command.inc(231):
_drush_invoke_hooks(Array, Array)
#7 /root/.composer/vendor/drush/drush/includes/command.inc(199):
drush_command('tripal')
#8
/root/.composer/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67):
drush_dispatch(Array)
#9 /root/.composer/vendor/drush/drush/includes/preflight.inc(67):
Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#10 /root/.composer/vendor/drush/drush/drush.php(12): drush_main()
#11 {main}
The command "docker exec -it tripal2 drush en -y tripal" exited with 1.
```

### Additional Notes

This issue is what's causing the errors in travis. See PR #903 for travis errors.

**Comment by bradfordcondon** [comment]
~~`tripal_custom_quota` should not already exist.   It isnt in the tripal 2 codebase at all, only in Tripal 3.~~ 

oops lacey is right.  I must not have pulled before checking the branch.

**Comment by laceysanderson** [comment]
It must have been backported? 
https://github.com/tripal/tripal/blob/7.x-2.x/tripal_core/api/tripal_core.quotas.api.inc
https://github.com/tripal/tripal/blob/7.x-2.x/tripal_core/tripal_core.install#L111

**Comment by almasaeed2010** [comment]
Addressed in #906 

Thanks all!
